include/stomp_moveit/cost_functions.hpp

```diff
@@ -96,5 +91,76 @@ CostFn get_collision_cost_function(const std::shared_ptr<const planning_scene::P
 
   return cost_fn;
 }
+}  // namespace
+namespace costs
+{
+// Interpolation step size for collision checking (joint space, L2 norm)
+constexpr double COL_CHECK_DISTANCE = 0.05;
+constexpr double CONSTRAINT_CHECK_DISTANCE = 0.1;
+
+CostFn get_collision_cost_function(const std::shared_ptr<const planning_scene::PlanningScene>& planning_scene,
+                                   const moveit::core::JointModelGroup* group, double collision_penalty)
+{
+  const auto& joints = group ? group->getActiveJointModels() : planning_scene->getRobotModel()->getActiveJointModels();
+  const auto& group_name = group ? group->getName() : "";
+
+  StateValidatorFn collision_validator_fn = [=](const Eigen::VectorXd& positions) {
+    static moveit::core::RobotState state(planning_scene->getCurrentState());
+
+    // Update robot state values
+    set_joint_positions(positions, joints, state);
+    state.update();
+
+    return !planning_scene->isStateColliding(state, group_name);
+  };
+
+  return get_cost_function_from_state_validator(collision_validator_fn, COL_CHECK_DISTANCE, collision_penalty);
+}
+
+CostFn get_constraints_cost_function(const std::shared_ptr<const planning_scene::PlanningScene>& planning_scene,
+                                     const moveit::core::JointModelGroup* group,
+                                     const moveit_msgs::msg::Constraints& constraints_msg, double constraints_penalty)
+{
+  const auto& joints = group ? group->getActiveJointModels() : planning_scene->getRobotModel()->getActiveJointModels();
+  const auto& group_name = group ? group->getName() : "";
+
+  kinematic_constraints::KinematicConstraintSet constraints_set(planning_scene->getRobotModel());
+  constraints_set.add(constraints_msg, planning_scene->getTransforms());
+
+  StateValidatorFn constraints_validator_fn = [=, constraints =
+                                                      std::move(constraints_set)](const Eigen::VectorXd& positions) {
+    static moveit::core::RobotState state(planning_scene->getCurrentState());
+
+    // Update robot state values
+    set_joint_positions(positions, joints, state);
+    state.update();
+
+    // NOTE: the returned ConstraintEvaluationResult also provides a `double distance` which might be used as an
+    // actual cost gradient instead of the binary state penalty
+    return constraints.decide(state).satisfied;
+  };
+
+  return get_cost_function_from_state_validator(constraints_validator_fn, CONSTRAINT_CHECK_DISTANCE,
+                                                constraints_penalty);
+}
+CostFn sum(const std::vector<CostFn>& cost_functions)
```

Comment:
Would it make sense to add here an optional vector with weights or a pair of cost function + weight as argument to be able to weight the cost functions differently?
So something like:
```cpp
overall_costs += costs * cost_function_weight.at(N)
```

Reply:
I think the penalty already takes care of that, but this is definitely a feature that we should consider when designing a cost API for MoveIt